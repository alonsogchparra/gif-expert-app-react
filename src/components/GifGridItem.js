import PropTypes from "prop-types";
import React from "react";

const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="col mb-4">
      <div className="card h-100 border-secondary animate__animated animate__fadeIn">
        <img className="card-img-top" src={url} alt={title} />
        <div className="card-body">
          <h4>{title}</h4>
        </div>
      </div>
    </div>
  );
};

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
