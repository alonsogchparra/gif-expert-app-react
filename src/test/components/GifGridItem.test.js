import { shallow } from "enzyme";
import React from "react";
import GifGridItem from "../../components/GifGridItem";

describe("test on GifGridItem", () => {
  const id = "ADBD3232";
  const title = "a title";
  const url = "https://localhost/anything";

  const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);

  test("should show GifGridItem correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should has h4 the title content", () => {
    expect(wrapper.find("h4").text().trim()).toBe(title);
  });

  test("should have the src equal to url and alt equal to title", () => {
    const image = wrapper.find("img");

    expect(image.prop("src")).toBe(url);
    expect(image.prop("alt")).toBe(title);
  });

  test("should have animate__fadeIn on className", () => {
    const div = wrapper.find("div").at(1);

    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
