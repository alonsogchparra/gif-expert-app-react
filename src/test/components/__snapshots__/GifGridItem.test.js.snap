// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifGridItem should show GifGridItem correctly 1`] = `
<div
  className="col mb-4"
>
  <div
    className="card h-100 border-secondary animate__animated animate__fadeIn"
  >
    <img
      alt="a title"
      className="card-img-top"
      src="https://localhost/anything"
    />
    <div
      className="card-body"
    >
      <h4>
        a title
      </h4>
    </div>
  </div>
</div>
`;
